Sources/Turf/Geometries/LineString.swift

```diff
@@ -215,4 +215,114 @@ extension LineString {
+                x += dx * t
+                y += dy * t
+            }
+        }
+
+        dx = coordinate.latitude - x
+        dy = coordinate.longitude - y
+
+        return dx * dx + dy * dy
+    }
+
+    private func simplifyDouglasPeuckerStep(_ coordinates: [CLLocationCoordinate2D], first: Int, last: Int, tolerance: Double, simplified: inout [CLLocationCoordinate2D]) {
+
+        var maxSquareDistance = tolerance
+        var index = 0
+
+        for i in first + 1 ..< last {
+            let squareDistance = getSquareSegmentDistance(coordinates[i], segmentStart: coordinates[first], segmentEnd: coordinates[last])
+
+            if squareDistance > maxSquareDistance {
+                index = i
+                maxSquareDistance = squareDistance
+            }
+        }
+
+        if maxSquareDistance > tolerance {
+            if index - first > 1 {
+                simplifyDouglasPeuckerStep(coordinates, first: first, last: index, tolerance: tolerance, simplified: &simplified)
+            }
+            simplified.append(coordinates[index])
+            if last - index > 1 {
+                simplifyDouglasPeuckerStep(coordinates, first: index, last: last, tolerance: tolerance, simplified: &simplified)
+            }
+        }
+    }
+
+    private func simplifyDouglasPeucker(_ coordinates: [CLLocationCoordinate2D], tolerance: Double!) -> [CLLocationCoordinate2D] {
+        if coordinates.count <= 2 {
+            return coordinates
+        }
+
+        let lastPoint = coordinates.count - 1
+        var result = [coordinates[0]]
+        simplifyDouglasPeuckerStep(coordinates, first: 0, last: lastPoint, tolerance: tolerance, simplified: &result)
+        result.append(coordinates[lastPoint])
+        return result
+    }
+
+    /// Returns a simplified version of the LineString using the Ramer–Douglas–Peucker algorithm.
+    ///
+    /// tolerance:  Controls the level of simplification by specifying the maximum allowed distance between the original line point
+    /// and the simplified point. Higher tolerance values results in higher simplification.
+    /// 
+    /// highestQuality: Excludes distance-based preprocessing step which leads to highest quality simplification. High quality simplification runs considerably slower so consider how much precision is needed in your application.
+    ///
+    /// Ported from https://github.com/Turfjs/turf/blob/master/packages/turf-simplify/lib/simplify.js
+    public func simplified(tolerance: Double = 1.0, highestQuality: Bool = false) -> LineString {
+        guard coordinates.count > 2 else { return LineString(coordinates) }
+
+        let squareTolerance = tolerance * tolerance
+
+        var simplifiedCoordinates = highestQuality ? coordinates : simplifyRadialDistance(coordinates, tolerance: squareTolerance)
+        
+        simplifiedCoordinates = simplifyDouglasPeucker(simplifiedCoordinates, tolerance: squareTolerance)
+
+        return LineString(simplifiedCoordinates)
```

Comment:
It would be convenient to also offer a mutable `simplify(tolerance:highestQuality:)` method that operates in place, to avoid the copying on this line, though such a method could be tail work.

Reply:
Added the mutable version as well.